src_features/provideErc20TokenInformation/cmd_provideTokenInfo.c

```diff
@@ -146,35 +62,43 @@ void handleProvideErc20TokenInformation(uint8_t p1,
     chain_id = U4BE(workBuffer, offset);
     if (!app_compatible_with_chain_id(&chain_id)) {
         UNSUPPORTED_CHAIN_ID_MSG(chain_id);
-        THROW(0x6A80);
+        THROW(APDU_RESPONSE_INVALID_DATA);
     }
     offset += 4;
     dataLength -= 4;
 
-#ifdef HAVE_TOKENS_EXTRA_LIST
-    tokenDefinition_t *currentToken = NULL;
-    uint32_t index;
-    for (index = 0; index < NUM_TOKENS_EXTRA; index++) {
-        currentToken = (tokenDefinition_t *) PIC(&TOKENS_EXTRA[index]);
-        if (memcmp(currentToken->address, token->address, 20) == 0) {
-            strcpy((char *) token->ticker, (char *) currentToken->ticker);
-            token->decimals = currentToken->decimals;
-            break;
+    PRINTF(
+        "[ERC20 Token Info] "
+        "=========================================================================================="
+        "=\n");
+#ifdef HAVE_LEDGER_PKI
+    error = os_pki_get_info(&key_usage, trusted_name, &trusted_name_len, &public_key);
+    if ((error == SWO_SUCCESS) && (key_usage == CERTIFICATE_PUBLIC_KEY_USAGE_COIN_META)) {
+        PRINTF(
+            "[ERC20 Token Info] Certificate '%s' loaded for usage "
+            "'CERTIFICATE_PUBLIC_KEY_USAGE_COIN_META'\n",
+            trusted_name);
+
+        // Checking the signature with PKI
+        if (!os_pki_verify(hash, sizeof(hash), (uint8_t *) (workBuffer + offset), dataLength)) {
```

Comment:
should be good to have a dedicated function to avoid code duplicate

Reply:
Done